imagersite/imager_images/tests.py

```diff
@@ -88,3 +79,326 @@ def test_add_photos_to_albums(self):
+    def setUp(self):
+        user = UserFactory.create(username='userbob')
+        user.set_password('secret')
+        user.save()
+        cover = PhotoFactory.create(user=user)
+        cover.save()
+        album = AlbumFactory.create(cover=cover, user=user)
+        album.save()
+        album.photos.add(cover)
+
+    def test_add_photo(self):
+        c = Client()
+        c.login(username='userbob', password='secret')
+        with open('imager_images/thumbnail.jpg', 'rb') as fh:
+            response = c.post(
+                '/images/photos/add/',
+                {'file': fh, 'title': 'test title', 'published': 'Private'},
+                follow=True
+            )
+        self.assertEqual(response.status_code, 200)
+        self.assertIn('<img src=\'/media/cache/', response.content)
+        self.assertIn('test title', response.content)
+
+    def test_photo_add_unauthenticated(self):
+        c = Client()
+        response = c.get('/images/photos/add', follow=True)
+        self.assertEqual(len(response.redirect_chain), 2)
+        self.assertIn('form method="post" action="/login/"',
+                      response.content)
+
+
+class PhotoEditTestCase(TestCase):
+    def setUp(self):
+        user = UserFactory.create(username='userbob')
+        user.set_password('secret')
+        user.save()
+        user2 = UserFactory.create(username='usereve')
+        user2.set_password('secret')
+        user2.save()
+        photo = PhotoFactory.create(user=user)
+        photo.save()
+
+    def test_edit_photo(self):
+        c = Client()
+        c.login(username='userbob', password='secret')
+        photo = Photo.objects.all()[0]
+        response = c.get('/images/photos/edit/{}/'.format(photo.id))
+        self.assertIn(photo.title, response.content)
+        with open('imager_images/thumbnail.jpg', 'rb') as fh:
+            response = c.post(
+                '/images/photos/edit/{}/'.format(photo.id),
+                {
+                    'file': fh,
+                    'title': 'new test title',
+                    'published': 'Private'
+                },
+                follow=True
+            )
+        self.assertIn('new test title', response.content)
+        response = c.get('/images/photos/{}/'.format(photo.id))
+        # make sure we have the same photo id
+        self.assertIn('new test title', response.content)
+
+    def test_edit_other_user(self):
+        # what you end up with is a create form for yourself.
+        c = Client()
+        c.login(username='usereve', password='secret')
+        photo = Photo.objects.all()[0]
+        with open('imager_images/thumbnail.jpg', 'rb') as fh:
+            response = c.post(
+                '/images/photos/edit/{}/'.format(photo.id),
+                {
+                    'file': fh,
+                    'title': 'other user',
+                    'published': 'Private'
+                },
+                follow=True
+            )
```

Comment:
given the number of times you do this, it might be the type of thing to make into a helper method.  `self.upload_photo(new=False)` or something like that.